corehq/apps/hqwebapp/models.py

```diff
@@ -35,6 +35,7 @@ class CommCareHQAlert(models.Model):
 
     text = models.TextField()
     domains = ArrayField(models.CharField(max_length=126), null=True)
+    created_by_domain = models.CharField(max_length=255, null=True, db_index=True)
```

Comment:
If we already have a nullable `domains` field, do we need a `created_by_domain` field?

We can filter global alerts using `domains__isnull=True`. In templates we can use `{{ alerts.domains|BOOL }}`.

Could we just allow an admin to manage alerts if they are a member of all domains in `domains`?

Reply:
Hello,

This is to identify if an alert was added "by" the domain and then let domain alter it.
You are right about filtering for global alerts, though we can't use the existing field to filter for alerts created by domain.
It is possible to superuser to add alerts for a domain as well, which populates the "domains" column with that domain.

> Could we just allow an admin to manage alerts if they are a member of all domains in domains?

Didn't think about that. Though if needed in future, current implementation can be extended for this.